server/server.cpp

```diff
@@ -23,15 +23,15 @@ std::vector<char> http_error_check(Http_Request* req, Metrics_Database* db) {
     Http_Response rep;
 
     int type = req->getType();
-    const char* metric = req->getMetric();
+    std::string metric(req->getMetric());
 
     if (type != Http_Request::GET){		
 	rep.Add_Header_Field("", "HTTP/1.1 400 Bad Request");
 	rep.Add_Body("");
 	return rep.Prepare_Http_Response();
     }
 
-    if ((strlen(metric) != strlen("/metrics")) || (strcmp(metric, "/metrics") != 0)) {		
+    if (metric.find("/metrics\0") == std::string::npos) {		
```

Comment:
I believe that "/metrics\0" is the same as "/metrics". I am guessing that the intent was to test whether the path ends with "/metrics". Have you tested that this works?